internal/pkg/scaffold/cmd_test.go

```diff
@@ -158,9 +158,17 @@ func main() {
 		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
 	}
 	// Create Service object to expose the metrics port(s).
-	_, err = metrics.CreateMetricsService(ctx, cfg, servicePorts)
+	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
 	if err != nil {
-		log.Info(err.Error())
+		log.Info("Could not create metrics Service: ", err.Error())
+	}
+
+	// Remove below code if you do not have Prometheus operator running in the cluster.
```

Comment:
Since we handle `ServiceMonitor` not being present, I'm not sure we should encourage developers to remove this. When the operator is eventually deployed, some clusters may have prometheus-operator that the developer isn't aware of when they're developing.

Reply:
I think we should at least leave a note that this is prom-operator specific resource. But agree, we should rephrase it, any suggestions?

Reply:
Yeah, agree that we can say this is for clusters that have `prometheus-operator`. Maybe:

> CreateServiceMonitors is useful when the operator is deployed in a cluster with the prometheus-operator because it will automatically create the Prometheus ServiceMonitor resources necessary to configure Prometheus to scrape metrics from this operator. If this operator is deployed to a cluster without a prometheus-operator deployment, it will return `ErrServiceMonitorNotPresent`, which can be used to safely skip ServiceMonitor creation.

Kind of wordy, but something along those lines?